pkg/resources/cloudconfig/configmap_test.go

```diff
@@ -97,26 +114,160 @@ func TestGetVsphereCloudConfig(t *testing.T) {
+			cluster: &kubermaticv1.Cluster{
+				Spec: kubermaticv1.ClusterSpec{
+					Version: *semver.NewSemverOrDie("v1.1.1"),
+					Cloud: kubermaticv1.CloudSpec{
+						Openstack: &kubermaticv1.OpenstackCloudSpec{},
+					},
+				},
+			},
+			dc: &kubermaticv1.Datacenter{
+				Spec: kubermaticv1.DatacenterSpec{
+					Openstack: &kubermaticv1.DatacenterSpecOpenstack{
+						UseOctavia: pointer.BoolPtr(false),
+					},
+				},
+			},
+			wantConfig: &openstack.CloudConfig{
+				LoadBalancer: openstack.LoadBalancerOpts{
+					LBVersion:  "v2",
+					LBMethod:   "ROUND_ROBIN",
+					UseOctavia: pointer.BoolPtr(false),
+				},
+				BlockStorage: openstack.BlockStorageOpts{
+					BSVersion: "auto",
+				},
+			},
+		},
+		{
+			name: "OpenStack use-octavia not set",
+			cluster: &kubermaticv1.Cluster{
+				Spec: kubermaticv1.ClusterSpec{
+					Version: *semver.NewSemverOrDie("v1.1.1"),
+					Cloud: kubermaticv1.CloudSpec{
+						Openstack: &kubermaticv1.OpenstackCloudSpec{
+							UseOctavia: pointer.BoolPtr(false),
+						},
+					},
+				},
+			},
+			dc: &kubermaticv1.Datacenter{
+				Spec: kubermaticv1.DatacenterSpec{
+					Openstack: &kubermaticv1.DatacenterSpecOpenstack{
+						UseOctavia: pointer.BoolPtr(true),
+					},
+				},
+			},
+			wantConfig: &openstack.CloudConfig{
+				LoadBalancer: openstack.LoadBalancerOpts{
+					LBVersion:  "v2",
+					LBMethod:   "ROUND_ROBIN",
+					UseOctavia: pointer.BoolPtr(false),
+				},
+				BlockStorage: openstack.BlockStorageOpts{
+					BSVersion: "auto",
+				},
+			},
+		},
+		{
+			name: "OpenStack use-octavia not set",
+			cluster: &kubermaticv1.Cluster{
+				Spec: kubermaticv1.ClusterSpec{
+					Version: *semver.NewSemverOrDie("v1.1.1"),
+					Cloud: kubermaticv1.CloudSpec{
+						Openstack: &kubermaticv1.OpenstackCloudSpec{},
+					},
+				},
+			},
+			dc: &kubermaticv1.Datacenter{
+				Spec: kubermaticv1.DatacenterSpec{
+					Openstack: &kubermaticv1.DatacenterSpecOpenstack{},
+				},
+			},
+			wantConfig: &openstack.CloudConfig{
+				LoadBalancer: openstack.LoadBalancerOpts{
+					LBVersion: "v2",
+					LBMethod:  "ROUND_ROBIN",
+				},
+				BlockStorage: openstack.BlockStorageOpts{
+					BSVersion: "auto",
+				},
 			},
 		},
 	}
 
 	for idx := range testCases {
 		tc := testCases[idx]
 		t.Run(tc.name, func(t *testing.T) {
-			cloudConfig, err := getVsphereCloudConfig(tc.cluster, tc.dc, resources.Credentials{})
+			cloudConfig, err := CloudConfig(tc.cluster, tc.dc, resources.Credentials{})
 			if err != nil {
 				t.Fatalf("Error trying to get cloudconfig: %v", err)
 			}
-			if err := tc.verify(cloudConfig); err != nil {
-				t.Error(err)
+			t.Logf("config: %v", cloudConfig)
+			var actual interface{}
+			switch {
+			case tc.cluster.Spec.Cloud.Openstack != nil:
+				actual = unmarshalINICloudConfig(t, &openstack.CloudConfig{}, cloudConfig)
+			case tc.cluster.Spec.Cloud.VSphere != nil:
+				actual = unmarshalINICloudConfig(t, &vsphere.CloudConfig{}, cloudConfig)
+			}
```

Comment:
This switch calls for 2 dedicated Test functions.

Reply:
Not sure I got your point. I'm providing a different structs `&vsphere.CloudConfig{}` or `&openstack.CloudConfig{}`. I'm using the same function as both Openstack and vSphere use an INI file. When we'll add test cases for other providers we'll use different test functions.

Reply:
I meant to have
`func TestGetVsphereCloudConfig(t *testing.T) {`
and
`func TestGetOpenstackCloudConfig(t *testing.T) {`

as a way to avoid empty `interface{}` and to provide each own test cases

Reply:
Ok, I see. I think that both ways are legitimate. Personally, I don't see any fundamental difference and any problem in using `interface{}` in this context. 

If we have dedicated tests things are more compartmented but we have some more repetitions. I'm not strongly opinioned on this, I think it's largely a matter of taste.

Reply:
@kron4eg As I had to do some changes anyway and I was neutral about this comment I split the tests per provider. Does it look better for you now?

Reply:
Yeah, now it's beautiful! 👍